### PR TITLE
Add a clarification to an error message

### PR DESCRIPTION
**Pull request by WhyNotHugo**
## Description

Makes this a bit more foolproof.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes https://github.com/qmk/qmk_firmware/issues/15202

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Needs a rebase.

**Comment by WhyNotHugo** [comment]
Rebased. Note this doesn't have any breaking changes anyway.

File: builddefs/message.mk [hunk 1]
```diff
@@ -87,6 +87,7 @@ define GENERATE_MSG_AVAILABLE_KEYMAPS
 endef
 MSG_AVAILABLE_KEYMAPS = $(eval $(call GENERATE_MSG_AVAILABLE_KEYMAPS))$(MSG_AVAILABLE_KEYMAPS_ACTUAL)
 
+MSG_BOOTLOADER_NOT_FOUND_BASE = Bootloader not found. Make sure the board is in DFU mode. See https://docs.qmk.fm/\#/newbs_flashing\n
```
Comment (fauxpark) [inline builddefs/message.mk:90]: Some bootloaders are not "DFU" bootloaders, this might be more confusing tbh.

Comment (WhyNotHugo) [reply builddefs/message.mk]: Updated wording. As you may have guesses from the nature of this PR, I'm a bit new to this (and the intent of this PR is to guide other newbies a bit better).

**Comment by WhyNotHugo** [comment]
Out of curiosity, why do some devices wait 0.5s, and other 5s between retries?

**Comment by drashna** [comment]
> Out of curiosity, why do some devices wait 0.5s, and other 5s between retries?

Timing issues, mostly.  caterina will disappear after 8 seconds, so 5s retry is too high.

**Review by drashna — APPROVED** [review]

**Comment by WhyNotHugo** [comment]
Anything blocking this merge?

**Review by tzarc — APPROVED** [review]

**Pull request merged**